src/main/java/CalcServlet.java

```diff
@@ -0,0 +1,143 @@
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.function.ToIntBiFunction;
+import java.util.function.ToIntFunction;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+@WebServlet("/calc")
+public class CalcServlet extends HttpServlet {
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+        resp.getWriter().println(highLevelSolve(
+                substituteVariables(req.getParameter("equation").replaceAll("\\s+", ""), req)
+        ));
+    }
+
+    private String substituteVariables(String equation, HttpServletRequest req) {
+        StringBuilder next = new StringBuilder(equation);
+
+        do {
+            equation = next.toString();
+
+            next = new StringBuilder();
+            for (int i = 0; i < equation.length(); ++i) {
+                if (Character.isAlphabetic(equation.codePointAt(i))) {
+                    next.append(req.getParameter(Character.toString(equation.charAt(i))));
+                } else {
+                    next.append(equation.charAt(i));
+                }
+            }
+        } while (!next.toString().equals(equation));
+
+        return next.toString();
+    }
+
+    private int highLevelSolve(String equation) {
+        int left;
+
+        do {
+            left = 0;
+
+            boolean opened = false;
+            StringBuilder next = new StringBuilder();
+            for (int i = 0; i < equation.length(); ++i) {
+                switch (equation.charAt(i)) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Switch statements should have a default label](https://app.codacy.com/manual/thejerome/IFMO_Servlets_20190915/pullRequest?prid=5134372)